disk-utils/mkswap.c

```diff
@@ -588,6 +616,13 @@ int main(int argc, char **argv)
 			ctl.offset = str2unum_or_err(optarg,
 					10, _("Invalid offset"), SINT_MAX(off_t));
 			break;
+		case 's':
+			ctl.size = str2unum_or_err(optarg,
```

Comment:
Please, strtosize_or_err() to make it possible to use `--size 100MiB` (and don't forget to add a note about possible suffixes to the man page (see another pages), we usually use
```  
       The size arguments may be followed by the multiplicative suffixes KiB (=1024), MiB
       (=1024*1024), and so on for GiB, TiB, PiB, EiB, ZiB, and YiB (the "iB" is optional, e.g., "K" has the
       same meaning as "KiB") 
```